Fix Issue #13: preferences.xhtml and preferences.js are missing in src-2.0

MischaMegens2 opened the pull request:
This puts back in the missing files, so the Settings pane functions again; moreover, it changes the color (normally red) according to the Settings (and plays nice with the 'Make It Green Instead' menu item).
Here is what it looks like:
<img width="1918" height="1124" alt="image" src="https://github.com/user-attachments/assets/9e605f9f-ab4f-4c4c-8395-beb500896a4f" />

MischaMegens2 committed:
Add preferences.xhtml supplying a preference pane

Puts back in preferences.xhtml and preferences.js, and adds extensions.make-it-red-text-color as a Pref in prefs.js. This makes the Make It Red preference pane in Settings work again.
The .xhtml is close to the one that's still available in https://zotero-download.s3.amazonaws.com/tmp/make-it-red/make-it-red-2.0.xpi, however, here:
- there is no <h2> title, this is not needed since Zotero already supplies a title
- it uses <html:input> instead of <input>, this way the color picker works (at least in Windows 11, for me)
- it comes with proper l10n

MischaMegens2 committed:
Make the style.css color a variable

This makes the style.css color a variable, and initializes it from the corresponding Settings. As a result, the Make It Red text color settings in the Settings pane actually have an effect.
(We have to make the green color a variable too to get valid CSS that plays nice with the menu item View/Make It Green Instead.)

MischaMegens2 committed:
Immediately update text color

After the previous commit, the plugin still needs to be disabled/re-enabled for new text color settings to take effect. This updates the color right away.

dawaltconley commented:
Thank you for this, I only learned how to properly set up preference panes by looking at this PR.

One note: this no longer works in Zotero 8 because [preference script variables are no longer in the global scope](https://groups.google.com/g/zotero-dev/c/uQhEGkJEzYs/m/QSHUZRf4BgAJ). This can be fixed by explicitly assigning global variables to the window object; i.e. changing line 1 of the `preferences.js` file from

```js
MakeItRed_Preferences = {
```
to
```js
window.MakeItRed_Preferences = {
```

That change is backwards-compatible with Zotero 7.

MischaMegens2 committed:
Update preferences.js, preference script variables no longer in global scope in Zotero 8

As suggested by @dawaltconley.
The change is backwards-compatible with Zotero 7.

MischaMegens2 commented:
Glad it was useful, and thank you for the suggested update!
